src/wasm/wasm-binary.cpp

```diff
@@ -5599,21 +5599,27 @@ bool WasmBinaryBuilder::maybeVisitRefTest(Expression*& out, uint32_t code) {
   if (code != BinaryConsts::RefTest) {
     return false;
   }
-  auto* curr = allocator.alloc<RefTest>();
-  WASM_UNREACHABLE("TODO (gc): ref.test");
-  curr->finalize();
-  out = curr;
+  auto heapType1 = getHeapType();
+  auto heapType2 = getHeapType();
+  auto* ref = popNonVoidExpression();
+  validateHeapTypeUsingChild(ref, heapType1);
+  auto* rtt = popNonVoidExpression();
+  validateHeapTypeUsingChild(rtt, heapType2);
```

Comment:
Hmm, I wonder if `validateChildUsingHeapType` would paint a clearer picture of what this does.

Reply:
I find that less clear actually. The child is already known to be valid at this point. The only question is whether the extra heap type immediate matches the child. But I don't feel strongly about the name.